In [4]:
# Load the required packages
import numpy as np, scipy as sc, matplotlib as plt, seaborn as sns, pandas as pd, statsmodels as sm, sklearn as sk

# Task 1: Source the Data Set
1. Import the Iris data set from the sklearn.datasets module.
2. Explain, in your own words, what the load_iris() function returns.

In [9]:
# Save the iris dataset in the iris object
iris = sk.datasets.load_iris()

In [20]:
iris.keys()

dict_keys(['data', 'target', 'frame', 'target_names', 'DESCR', 'feature_names', 'filename', 'data_module'])

The load_iris() function returns a dictionary-like object which contains multiple obkects, including: the iris dataset and the metadata of the dataset. This keys of this object are 'data', 'target', 'frame', 'target_names', 'DESCR', 'feature_names', 'filename', 'data_module'.

# Task 2: Explore the Data Structure
Print and explain the shape of the data set, the first and last 5 rows of the data, the feature names, and the target classes.

In [21]:
# Display the shape of the Iris data
iris.data.shape

(150, 4)

The Iris dataset contains 150 rows (each sample is in a separate row) and 4 collums (each collumn corresponds to sepal length, sepal width, petal length and petal width).

In [24]:
# Display the first 5 rows
iris.data[0:5,]

array([[5.1, 3.5, 1.4, 0.2],
       [4.9, 3. , 1.4, 0.2],
       [4.7, 3.2, 1.3, 0.2],
       [4.6, 3.1, 1.5, 0.2],
       [5. , 3.6, 1.4, 0.2]])

The dataset above contains info about sepal length, sepal width, petal length and petal width for the first 5 samples in the dataset.

In [32]:
# Display the last five rows
iris.data[-5:len(iris.data),]

array([[6.7, 3. , 5.2, 2.3],
       [6.3, 2.5, 5. , 1.9],
       [6.5, 3. , 5.2, 2. ],
       [6.2, 3.4, 5.4, 2.3],
       [5.9, 3. , 5.1, 1.8]])

The dataset above contains info about sepal length, sepal width, petal length and petal width for the last 5 samples in the dataset.

In [33]:
# Display the feature names
iris.feature_names

['sepal length (cm)',
 'sepal width (cm)',
 'petal length (cm)',
 'petal width (cm)']

The feature_names object gives the explanation what does each collumn in the dataset represents.

In [35]:
# Display the sample types in the dataset
iris.target_names

array(['setosa', 'versicolor', 'virginica'], dtype='<U10')

The target_names shows what sample types are present in the dataset.

# Task 3: Summarize the Data
For each feature in the dataset, calculate and display:  
**mean  
minimum  
maximum  
standard deviation  
median**  

In [51]:
# Calculate the mean of the features
def iris_mean(dataset, feature_number):
    # Count the number of samples
    count = len(dataset)
    
    total_feature_val = 0
    
    # Loop through each sample and add the value of a selected feature to the total_feature_val
    for sample in dataset:
        total_feature_val += sample[feature_number]
    
    # Calculate the mean of the feature
    mean = total_feature_val / count
    
    # Round the mean to the second decimel place after the dot
    mean = round(mean,2)
    
    return(mean)

np.float64(5.84)

In [53]:
sepal_length = iris_mean(iris.data, 0)
sepal_width = iris_mean(iris.data, 1)
petal_lenght = iris_mean(iris.data, 2)
petal_width = iris_mean(iris.data, 3)

print(
    f"""
sepal length (cm): {sepal_length}
sepal width (cm): {sepal_width}
petal length (cm): {petal_lenght}
petal width (cm): {petal_width}
""")


sepal length (cm): 5.84
sepal width (cm): 3.06
petal length (cm): 3.76
petal width (cm): 1.2



In [70]:
# Separate data for each feature and convert it to numpy array
def separate_features(dataset, feature_number):
    
    # Separate a feature
    feature_data = [int(sample[feature_number]) for sample in dataset]
    
    # Convert the integer list to numpy array
    feature_data = np.array(feature_data)
    
    return(feature_data)

In [71]:
sepal_length_data = separate_features(iris.data, 0)
sepal_width_data = separate_features(iris.data, 1)
petal_lenght_data = separate_features(iris.data, 2)
petal_width_data = separate_features(iris.data, 3)

In [74]:
# Calculate standard deviations of each dataset using the inherent numpy.array function sdev
sdt_sepal_length = round(sepal_length_data.std(), 3)
sdt_sepal_width = round(sepal_width_data.std(), 3)
sdt_petal_lenght = round(petal_lenght_data.std(), 3)
sdt_petal_width = round(petal_width_data.std(), 3)

print(
    f"""
Standard deviation of sepal length (cm): {sdt_sepal_length}
Standard deviation of sepal width (cm): {sdt_sepal_width}
Standard deviation of petal length (cm): {sdt_petal_lenght}
Standard deviation of petal width (cm): {sdt_petal_width}
""")


Standard deviation of sepal length (cm): 0.839
Standard deviation of sepal width (cm): 0.531
Standard deviation of petal length (cm): 1.766
Standard deviation of petal width (cm): 0.712



In [75]:
# Calculate maximum of each dataset using the inherent numpy.array function max
max_sepal_length = sepal_length_data.max()
max_sepal_width = sepal_width_data.max()
max_petal_lenght = petal_lenght_data.max()
max_petal_width = petal_width_data.max()

print(
    f"""
Maximum of sepal length (cm): {max_sepal_length}
Maximum of sepal width (cm): {max_sepal_width}
Maximum of petal length (cm): {max_petal_lenght}
Maximum of petal width (cm): {max_petal_width}
""")


Maximum of sepal length (cm): 7
Maximum of sepal width (cm): 4
Maximum of petal length (cm): 6
Maximum of petal width (cm): 2



In [76]:
# Calculate minimum of each dataset using the inherent numpy.array function min
min_sepal_length = sepal_length_data.min()
min_sepal_width = sepal_width_data.min()
min_petal_lenght = petal_lenght_data.min()
min_petal_width = petal_width_data.min()

print(
    f"""
Minimum of sepal length (cm): {min_sepal_length}
Minimum of sepal width (cm): {min_sepal_width}
Minimum of petal length (cm): {min_petal_lenght}
Minimum of petal width (cm): {min_petal_width}
""")


Minimum of sepal length (cm): 4
Minimum of sepal width (cm): 2
Minimum of petal length (cm): 1
Minimum of petal width (cm): 0



In [80]:
# Calculate median of each dataset using the inherent numpy.array function median
median_sepal_length = np.median(sepal_length_data)
median_sepal_width = np.median(sepal_width_data)
median_petal_lenght = np.median(petal_lenght_data)
median_petal_width = np.median(petal_width_data)

print(
    f"""
Median of sepal length (cm): {median_sepal_length}
Median of sepal width (cm): {median_sepal_width}
Median of petal length (cm): {median_petal_lenght}
Median of petal width (cm): {median_petal_width}
""")


Median of sepal length (cm): 5.0
Median of sepal width (cm): 3.0
Median of petal length (cm): 4.0
Median of petal width (cm): 1.0

